shell/components/AppModal.vue

```diff
@@ -59,12 +79,15 @@ export default Vue.extend({
     >
       <div
         class="modal-overlay"
+        :data-modal="name"
         @click="handleClickOutside"
       >
         <div
+          v-bind="$attrs"
           ref="modalRef"
+          :class="customClass"
           class="modal-container"
-          :style="{ width: width }"
+          :style="{ width: modalWidth, ...styles }"
```

Comment:
Yeah, I think we can use a validator for the `width` prop to help out here.

Reply:
I was thinking more about `styles`, but does not have to be done now tbh.

Reply:
I see, that's a good point. I've made an update to this so that we parse the `styles` string down to the object syntax [^1]. This should help, lmk what you think!

Reply:
I meant more evaluating the property to be valid, but it's fair enough like this